go-controller/hybrid-overlay/cmd/hybrid-overlay-node/hybrid-overlay-node.go

```diff
@@ -38,40 +39,52 @@ func main() {
 			Destination: &nodeName,
 		},
 		&cli.BoolFlag{
-			Name:        "windows-service",
-			Usage:       "Enables hybrid overlay to run as a Windows service. Ignored on Linux.",
-			Destination: &runAsWindowsService,
+			Name:  "windows-service",
+			Usage: "Enables hybrid overlay to run as a Windows service. Ignored on Linux.",
 		}})
+
+	for _, flag := range c.Flags {
```

Comment:
I would actually just do this from the windows implementation of initForOS(). Pass the `*cli.Context` to initForOS() and then you can use ctx.Bool() as a shortcut to get the flag value, like so:

```
const windowsServiceArgName string = "windows-service"

func initForOS(ctx *context.Context) error {
	if ctx.Bool(windowServiceArgName) {
		klog.Infof("Initializing Windows service")
		return service.InitService(appName, ctx)
	}
	return nil
}
```
and also use the constant windowsServiceArgName in the flag definition in hybrid-overlay-node.go.

Reply:
Then make the signal handling stuff that's run for Linux and not-windows-service a function in hybrid-overlay-node.go, and call that function from both the linux initForOS() and the not-windows-service block in Windows initForOS() implementation.

Reply:
thanks for the review @dcbw  I have added  a commit to address the changes you have suggested PTAL.